gcovr/workers.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding:utf-8 -*-
+
+# This file is part of gcovr <http://gcovr.com/>.
+#
+# Copyright 2013-2018 the gcovr authors
+# Copyright 2013 Sandia Corporation
+# This software is distributed under the BSD license.
+
+
+from threading import Thread, Condition
+from contextlib import contextmanager
+
+import sys
+if sys.version_info[0] >= 3:
+    from queue import Queue, Empty
+else:
+    from Queue import Queue, Empty
+
+
+class LockedDirectories(object):
+    """
+    Class that keeps a list of locked directories
+    """
+    def __init__(self):
+        self.dirs = set()
+        self.cv = Condition()
+
+    def run_in(self, dir_):
+        """
+        Start running in the directory and lock it
+        """
+        self.cv.acquire()
+        while dir_ in self.dirs:
+            self.cv.wait()
+        self.dirs.add(dir_)
+        self.cv.release()
+
+    def done(self, dir_):
+        """
+        Finished with the directory, unlock it
+        """
+        self.cv.acquire()
+        self.dirs.remove(dir_)
+        self.cv.notify()
```

Comment:
This shall be `self.cv.notify_all()` instead of just `self.cv.notify()`

Reply:
@mayeut I see why... that might explain the Windows freeze that appveyor is experiencing - because the right thread isn't being woken and we're deadlocking. Sound right?

Reply:
@mayeut It certainly works now whereas it didn't before!